### PR TITLE
Auth finalization

### PR DESCRIPTION
**Pull request by Anthonyp0329**
What changes did you make?

Fledged out all the auth features.

### Testing

How did you confirm your changes worked?

Ran everything, tried different login flows. 

### Confirmation of Change

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

TODO...
Still need to add logout route

### CODE REVIEW

File: SDCTA/src/components/auth/SignUpPage.tsx [hunk 1]
```diff
@@ -24,149 +14,15 @@ export const SignUpPage: React.FC = () => {
...
-        //         // goToApp();
-        //         navigate("/");
-        //       });
-        //     });
-        //   });
-        // }
-      }); // end of catch
+  const callSignupWithGoogle = (): void => {
+    dispatch({
+      type: "SIGNUP_GOOGLE_USER",
```
Comment (NirmalAgnihotri) [inline SDCTA/src/components/auth/SignUpPage.tsx:21]: We probably want action strings kept in a constants file or more likely just the file where the slice is defined

File: SDCTA/src/pages/SignUpDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { DetailsPage } from "../components/auth/DetailsPage";
+
+export const SignUpDetails: React.FC = () => {
+  return (
+    <div>
+      <DetailsPage />
```
Comment (NirmalAgnihotri) [inline SDCTA/src/pages/SignUpDetails.tsx:6]: We might not need this wrapping, but we can just rename DetailsPage to SignUpDetailsPage or something to make it less general and clear that its associated with the SignUpDetails

File: server/routes/user.js [hunk 3]
```diff
@@ -21,7 +21,9 @@ router.post("/addCard", async (req, res, next) => {
 // Create User Post Method
 router.post("/", async (req, res, next) => {
   try {
+    console.log("here");
```
Comment (NirmalAgnihotri) [inline server/routes/user.js:24]: We should remove this

File: SDCTA/src/sagas/authSaga.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import { auth } from "../firebase-config";
+import { deleteUser } from "firebase/auth";
+import {
+  register,
+  loginUser,
+  getUser,
+  signupWithGoogle,
+  type GoogleLogInReturn,
+} from "../api/auth";
+/*
```
Comment (NirmalAgnihotri) [inline SDCTA/src/sagas/authSaga.ts:22]: comment not needed

**Review by NirmalAgnihotri — APPROVED** [review]

**Review by NirmalAgnihotri — APPROVED** [review]

File: SDCTA/src/sagas/exampleSaga.ts [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
+// import { call, put, takeEvery, takeLatest } from "redux-saga/effects";
 
-// The following is an example of a worker saga from https://redux-saga.js.org/docs/introduction/GettingStarted
-// We will import the appropriate API functions that call our backend here and then use them in the Saga.
-// Note that in our case, we don't explicitly create the action type names since they are created for us by Redux toolkit
-// An action type can be obtained by calling the action creator and appending .type Ex: long().type
+// // The following is an example of a worker saga from https://redux-saga.js.org/docs/introduction/GettingStarted
+// // We will import the appropriate API functions that call our backend here and then use them in the Saga.
+// // Note that in our case, we don't explicitly create the action type names since they are created for us by Redux toolkit
+// // An action type can be obtained by calling the action creator and appending .type Ex: long().type
```
Comment (hpai1) [inline SDCTA/src/sagas/exampleSaga.ts:6]: File is full of comments?

**Review by emmaz12 — APPROVED** [review]

**Pull request merged**